src/sidebar/zeplinComponent/data/CumulativeBarrelDetailsStore.ts

```diff
@@ -0,0 +1,52 @@
+import BarrelDetails from "../../../common/domain/zeplinComponent/model/BarrelDetails";
+import Store from "../../../common/domain/store/Store";
+import BarrelError from "../../../common/domain/zeplinComponent/model/BarrelError";
+import BarrelType from "../../../common/domain/barrel/BarrelType";
+import Result from "../../../common/domain/store/Result";
+import BarrelDetailsStoreProvider from "../../../common/domain/zeplinComponent/data/BarrelDetailsStoreProvider";
+import { convertBarrelResult } from "../../../common/domain/barrel/util/barrelUtil";
+
+export default class CumulativeBarrelDetailsStore implements Store<BarrelDetails[], BarrelError> {
+    public constructor(private barrelId: string, private barrelType: BarrelType) { }
+
+    public get = async (): Promise<Result<BarrelDetails[], BarrelError>> => {
+        const leafId = this.barrelId;
+        const leafType = this.barrelType;
+        let currentId: string | undefined = leafId;
+        let currentType = leafType;
+        const accumulatedResult: Result<BarrelDetails[], BarrelError> = {
+            data: [],
+            errors: []
+        };
+
+        while (currentId) {
+            const childId: string | undefined = currentId === leafId ? undefined : leafId;
+            const childType: BarrelType | undefined = currentId === leafId ? undefined : leafType;
+
+            const { data, errors }: Result<BarrelDetails, BarrelError> = await BarrelDetailsStoreProvider
+                .get(currentId, currentType, childId, childType)
+                .get()
+                .then(
+                    barrel => convertBarrelResult(barrel, currentType, currentId!) // eslint-disable-line no-loop-func
+                );
+
+            if (errors) {
+                accumulatedResult.errors!.push(...errors);
+                currentId = undefined; // eslint-disable-line require-atomic-updates
```

Comment:
I can not see any race conditions here? Am I missing something?

Reply:
What do you mean?

Reply:
Why did you need to use `eslint-disable-line require-atomic-updates`?

Reply:
Ah, you are right, I'll remove them, probably added while I was trying something